solana/programs/token-bridge-relayer/src/processor/peers.rs

```diff
@@ -64,77 +57,54 @@ pub fn register_peer(
 ) -> Result<()> {
     // If it is the first peer for this chain, make it canonical:
```

Comment:
Just to be safe, we should probably check that the chain id isn't 0 or Solana and that the peer isn't the zero address.